homeassistant/components/vacuum/__init__.py

```diff
@@ -282,10 +347,22 @@ def state_attributes(self) -> dict[str, Any]:
 
         return data
 
-    @cached_property
+    @final
+    @property
     def state(self) -> str | None:
         """Return the state of the vacuum cleaner."""
-        return self._attr_state
+        if (vacuum_state := self.vacuum_state) is None:
+            return None
+        return vacuum_state
+
+    @cached_property
+    def vacuum_state(self) -> VacuumState | None:
```

Comment:
I think we decided to call this `activity` similar to lawn mower.

Reply:
Has been changed accordingly